### PR TITLE
moving lmod to the configuration file

### PR DESCRIPTION
**Pull request by Galithil**
Apparently, you can't use lmod like this on Irma. Therefore, moving the path to the configuration file allows any lmod to be used.

### CODE REVIEW

**Comment by sylvinite** [comment]
@vezzi @b97pla Looks good?

**Comment by b97pla** [comment]
The code looks sane to me but I must admit I don't know what `lmod` and this code do..

**Comment by sylvinite** [comment]
When ngi_pipeline tries to load modules (i.e. fastqc for the qc pipeline) it checks the lmod directory (which contains lua scripts) in order to determine where the module is located. 
Since irma has a bit differing folder structure from nestor, this caused ngi_pipeline_start.py qc to crash whenever it was started.

**Comment by Galithil** [comment]
lmod is involved in the module loading on Uppmax servers. Apaprently, on
Irma, it's not there, so the point is to allow someone to specify another
path to lmod, while keeping the current as default.

**Comment by vezzi** [comment]
Ok, can we also contact Uppmax and point them to this problem?

**Comment by johanherman** [comment]
lmod is on all UPPMAX systems. 

[johanhe@nestor1 ~]$ ls -la `env | grep LMOD_CMD | cut -f2 -d=`
-rwxr-xr-x 1 root root 25652 21 aug  2013 /usr/lib/lmod/lmod/libexec/lmod

[johanhe@irma1 johanhe]$ ls -la `env | grep LMOD_CMD | cut -f2 -d=`
-rwxr-xr-x 1 root root 24456 28 okt 20.29 /usr/share/lmod/lmod/libexec/lmod

**Comment by Galithil** [comment]
We're not saying irma is missing lmod, we're saying it's not in the same
place, so we need to update the code.

**Comment by johanherman** [comment]
So can't you just do an `os.environ('LMOD_CMD')` or something similar? Or am I missing something?

**Comment by Galithil** [comment]
It should be possible. Is that also available on submitted SLURM jobs ?

**Comment by johanherman** [comment]
It seems to be on Irma at least: 

```
[johanhe@irma1 tmp]$ cat testbatch.sh 
#!/bin/bash -l
 
#SBATCH -A ngi2016001
#SBATCH -p core
#SBATCH -n 1
#SBATCH -t 12:00:00
#SBATCH -J testjob

echo $LMOD_CMD > /home/johanhe/tmp/lmod-out.txt

[johanhe@irma1 tmp]$ sbatch testbatch.sh 
Submitted batch job 3778

[johanhe@irma1 tmp]$ cat lmod-out.txt 
/usr/share/lmod/lmod/libexec/lmod
```
I presume it is the same on other systems.

File: ngi_pipeline/utils/filesystem.py [hunk 1]
```diff
@@ -23,7 +23,8 @@
 
 LOG = minimal_logger(__name__)
 
-def load_modules(modules_list):
+@with_ngi_config
+def load_modules(modules_list, config=None, config_file_path=None):
```
Comment (johanherman) [inline ngi_pipeline/utils/filesystem.py]: I'm not sure about your coding conventions, but perhaps having docstrings for `config` and `config_file_path`?

**Comment by sylvinite** [comment]
Jumping the gun and merging this. After some P2P talks no one seems to have any problems with this.

**Pull request merged**